src/qt/diagnosticsdialog.cpp

```diff
@@ -9,27 +9,148 @@
 
 #include "diagnosticsdialog.h"
 #include "ui_diagnosticsdialog.h"
+#include "neuralnet/researcher.h"
+#include "upgrade.h"
 
 #include <numeric>
 #include <fstream>
 
+extern std::unique_ptr<Upgrade> g_UpdateChecker;
+
```

Comment:
```suggestion
```

This is already declared by including upgrade.h.

Reply:
Yep.